src/crypto/tls/common.go

```diff
@@ -674,6 +718,23 @@ type Config struct {
 	// used for debugging.
 	KeyLogWriter io.Writer
 
+	// ECHEnabled determines whether the ECH extension will be used for this
+	// connection.
+	ECHEnabled bool
```

Comment:
If ECHEnabled is set but there are no configurations to use, then a dummy ECH extension is sent instead. Note that we can't send dummy ECH by default, since this would break existing unittests.

Reply:
Ah, gotcha. This is for GREASE. Thanks!